src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1308,6 +1308,48 @@ bool QgsOgrProvider::deleteAttributes( const QgsAttributeIds &attributes )
 #endif
 }
 
+bool QgsOgrProvider::renameAttributes( const QgsFieldNameMap& renamedAttributes )
+{
+  if ( !doInitialActionsForEdition() )
+    return false;
+
+#if defined(GDAL_VERSION_NUM) && GDAL_VERSION_NUM >= 1900
+  QgsFieldNameMap::const_iterator renameIt = renamedAttributes.constBegin();
+  bool result = true;
+  for ( ; renameIt != renamedAttributes.constEnd(); ++renameIt )
+  {
+    int fieldIndex = renameIt.key();
+    if ( fieldIndex < 0 || fieldIndex >= mAttributeFields.count() )
```

Comment:
There might be a shift of 1 between OGR field index and QGIS field index if mFirstFieldIsFid is TRUE.   When mFirstFieldIsFid is TRUE, ogr field index = qgis field index - 1 for qgis field index >= 1. And if qgis field index =  0 (OGR FID column), then it should probably error out.